tests/compile-and-dump/Promote/GenDefunSymbols.ghc88.template

```diff
@@ -9,7 +9,7 @@ Promote/GenDefunSymbols.hs:0:0:: Splicing declarations
       where
         LiftMaybeSym1KindInference :: forall f0123456789876543210
                                              x0123456789876543210
-                                             arg. Data.Singletons.Internal.SameKind (Apply (LiftMaybeSym1 f0123456789876543210) arg) (LiftMaybeSym2 f0123456789876543210 arg) =>
+                                             arg. singletons-2.6:Data.Singletons.Internal.SameKind (Apply (LiftMaybeSym1 f0123456789876543210) arg) (LiftMaybeSym2 f0123456789876543210 arg) =>
```

Comment:
I doubt that this is directly the result of your changes, but it's a bit worrying that the `singletons-2.6` tag is creeping into test output like this, since it means that we'll have to update golden files every time the version number changes. I thought that [this code](https://github.com/goldfirere/singletons/blob/ea5787091689e857ea7c71a89ac2823644e493bd/tests/SingletonsTestSuiteUtils.hs#L175-L178) was stripping out the version numbers—do you have any idea why it isn't doing so now?

Reply:
> do you have any idea why it isn't doing so now?

I don't know if I'm running the test suite correctly, the command I use is `cabal v2-test --allow-newer`. Could that be the problem?

Reply:
That's exactly how I run the test suite as well, so I doubt that that's the problem.

Reply:
Seems to be a local misconfiguration, on CI the `singletons-2.6` tag is stripped off: https://travis-ci.org/goldfirere/singletons/jobs/525356545

Reply:
Interesting! I wonder if perhaps this is caused by your machine having a different version of `sed`? As one example, we currently have to employ [some CPP](https://github.com/goldfirere/singletons/blob/ea5787091689e857ea7c71a89ac2823644e493bd/tests/SingletonsTestSuiteUtils.hs#L163-L167) to work around differences between the Linux and macOS versions of `sed`.

Reply:
macOS `sed` makes me sad. I solved the issue by using `sed` from `nixpkgs`

Reply:
Bah, so it _is_ a macOS issue. In that case, can you open a separate issue about this? I really hate having platform-dependent behavior affecting whether the test suite passes, so perhaps we can replace the use of `sed` with something that has more predictable behavior.

Reply:
I created #391